color/src/color.rs

```diff
@@ -0,0 +1,531 @@
+        let x = self.components;
+        let y = rhs.components;
+        Self::new([x[0] + y[0], x[1] + y[1], x[2] + y[2]])
+    }
+}
+
+impl<T: Colorspace> core::ops::Sub for OpaqueColor<T> {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self {
+        let x = self.components;
+        let y = rhs.components;
+        Self::new([x[0] - y[0], x[1] - y[1], x[2] - y[2]])
+    }
+}
+
+impl<T: Colorspace> core::ops::Mul<f32> for AlphaColor<T> {
+    type Output = Self;
+
+    fn mul(self, rhs: f32) -> Self {
+        Self::new(self.components.map(|x| x * rhs))
+    }
+}
+
+impl<T: Colorspace> core::ops::Mul<AlphaColor<T>> for f32 {
+    type Output = AlphaColor<T>;
+
+    fn mul(self, rhs: AlphaColor<T>) -> Self::Output {
+        rhs * self
+    }
+}
+
+impl<T: Colorspace> core::ops::Div<f32> for AlphaColor<T> {
+    type Output = Self;
+
+    #[expect(
+        clippy::suspicious_arithmetic_impl,
+        reason = "somebody please teach clippy about multiplicative inverses"
+    )]
+    fn div(self, rhs: f32) -> Self {
+        self * rhs.recip()
+    }
+}
+
+impl<T: Colorspace> core::ops::Add for AlphaColor<T> {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        let x = self.components;
+        let y = rhs.components;
+        Self::new([x[0] + y[0], x[1] + y[1], x[2] + y[2], x[3] + y[3]])
+    }
+}
+
+impl<T: Colorspace> core::ops::Sub for AlphaColor<T> {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self {
+        let x = self.components;
+        let y = rhs.components;
+        Self::new([x[0] - y[0], x[1] - y[1], x[2] - y[2], x[3] - y[3]])
+    }
+}
+
+impl<T: Colorspace> core::ops::Mul<f32> for PremulColor<T> {
+    type Output = Self;
+
+    fn mul(self, rhs: f32) -> Self {
+        Self::new(self.components.map(|x| x * rhs))
+    }
+}
+
+impl<T: Colorspace> core::ops::Mul<PremulColor<T>> for f32 {
+    type Output = PremulColor<T>;
+
+    fn mul(self, rhs: PremulColor<T>) -> Self::Output {
+        rhs * self
+    }
+}
+
+impl<T: Colorspace> core::ops::Div<f32> for PremulColor<T> {
+    type Output = Self;
+
+    #[expect(
+        clippy::suspicious_arithmetic_impl,
+        reason = "somebody please teach clippy about multiplicative inverses"
+    )]
+    fn div(self, rhs: f32) -> Self {
+        self * rhs.recip()
+    }
+}
+
+impl<T: Colorspace> core::ops::Add for PremulColor<T> {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        let x = self.components;
+        let y = rhs.components;
+        Self::new([x[0] + y[0], x[1] + y[1], x[2] + y[2], x[3] + y[3]])
+    }
+}
+
+impl<T: Colorspace> core::ops::Sub for PremulColor<T> {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self {
+        let x = self.components;
+        let y = rhs.components;
+        Self::new([x[0] - y[0], x[1] - y[1], x[2] - y[2], x[3] - y[3]])
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::{fixup_hue, HueDirection};
+
+    #[test]
+    fn test_hue_fixup() {
```

Comment:
I think that this `test_` prefix is unidiomatic. See e.g. https://github.com/rust-lang/rust-clippy/issues/8931